core/src/main/java/org/apache/accumulo/core/clientImpl/ZookeeperLockChecker.java

```diff
@@ -59,11 +59,8 @@ public boolean isLockHeld(String server, String session) {
 
   @Override
   public void invalidateCache(String tserver) {
-    var hostAndPort = HostAndPort.fromString(tserver);
-    ctx.getServerPaths().getTabletServer(rg -> true, AddressSelector.exact(hostAndPort), false)
-        .forEach(slp -> {
-          ctx.getZooCache().clear(slp.toString());
-        });
+    // The path for the tserver contains a resource group. The resource group is unknown, so can not
+    // construct a prefix. Therefore clear any path that contains the tserver.
+    ctx.getZooCache().clear(path -> path.contains(tserver));
```

Comment:
I'm wondering if this could overmatch, where the tserver is also in another path, like dead tservers or something. I wonder if we want the predicate to be:
```
path.startsWith(ctx.getZooKeeperRoot() + Constants.ZTSERVERS) && path.contains(tserver)
```

Reply:
Definitely want to avoid overmatching as that could cause performance problems.  Will make this change before merging.

Reply:
updated in c3b3e64